ckan/tests/lib/test_mailer.py

```diff
@@ -34,6 +34,40 @@ def get_email_subject(self, msg):
 class TestMailer(MailerBase):
     def test_mail_recipient(self, mail_server):
         user = factories.User()
+        config['ckan.hide_version'] = "False"
+
+        msgs = mail_server.get_smtp_messages()
+        assert msgs == []
+
+        # send email
+        test_email = {
+            "recipient_name": "Bob",
+            "recipient_email": user["email"],
+            "subject": "Meeting",
+            "body": "The meeting is cancelled.\n",
+            "headers": {"header1": "value1"},
+        }
+        mailer.mail_recipient(**test_email)
+
+        # check it went to the mock smtp server
+        msgs = mail_server.get_smtp_messages()
+        assert len(msgs) == 1
+        msg = msgs[0]
+        assert msg[1] == config["smtp.mail_from"]
+        assert msg[2] == [test_email["recipient_email"]]
+        assert list(test_email["headers"].keys())[0] in msg[3], msg[3]
+        assert list(test_email["headers"].values())[0] in msg[3], msg[3]
+        assert test_email["subject"] in msg[3], msg[3]
+        assert "X-Mailer" in msg[3], "Missing X-Mailer header"
+        expected_body = self.mime_encode(
+            test_email["body"], test_email["recipient_name"]
+        )
+        assert expected_body in msg[3]
+
+    def test_mail_recipient_hiding_mailer(self, mail_server):
+        user = factories.User()
+        config['ckan.hide_version'] = "True"
```

Comment:
Use the `@pytest.mark.ckan_config("ckan.hide_version", "True")` decorator instead of changing the `config` object directly